Fix/protractor implementation

SkxPhan opened the pull request:
Pull request to fix issue #1.

1. Fix resample function to ensure correct number of points
2. Fix vectorizeStroke function based on Protractor pseudo code paper
3. Fix optimalCosineDistance function
4. Adapt and add test case for the _recognize_ function

SkxPhan committed:
Fix resample function to ensure correct number of points

SkxPhan committed:
Fix vectorizeStroke function based on Protractor pseudo code paper

SkxPhan committed:
Fix optimalCosineDistance function.
- Update function to accommodate new version of vectorizeStroke.
- Correct sign error in calculation.
- Remove unused vecCrossProduct function.

SkxPhan committed:
Adapt and add test case for recognize function

SkxPhan committed:
Remove vs code file

SkxPhan committed:
Remove redeclaration of the vector records

htfy96 reviewed (approved):
Sorry for the delay. Checked all proposed changes and they all look great. Thanks for your contribution!

htfy96 commented on test/test.cpp:
```diff
@@ -13,7 +13,17 @@ TEST_CASE("1 instance")
+    Stroke testStroke{ { { 0., 0. }, { 0., 0.9 }, { 0.02, 0.91 }, { 1., 1. } }, Orientation::Sensitive };
+    auto [it, score] = recognize(testStroke, strokes.begin(), strokes.end());
+    CHECK(it == strokes.begin());
+    CHECK_THAT(score, Catch::Matchers::WithinAbs(17.6808f, 0.001f));
```
This threshold is too stringent and caused the test to fail in Github CI. Can you change it to `0.01f`?

SkxPhan replied:
Done!

SkxPhan committed:
Fix too stringent threshold for Github CI.

SkxPhan committed:
Fix too stringent threshold for Github CI

htfy96 commented:
Hi, the CIs are still failing. Would you like to a look when having time? Alternatively I can also help debugging

SkxPhan commented:
Hi, I think the first test failed because of the perfect match. Therefore score reach an infinite value due to a zero division happening in the function _recognize_:
```cpp
recognize(const Stroke& target, ItT templateItBegin, ItT templateItEnd)
{
    float maxScore = 0;
    ItT matchedIt = templateItEnd;
    for (ItT templateIt = templateItBegin; templateIt != templateItEnd; ++templateIt) {
        const float d = detail::optimalCosineDistance(target.getVectorizedStroke(), templateIt->getVectorizedStroke());
        const float score = 1.0f / d;
        if (score > maxScore) {
            maxScore = score;
            matchedIt = templateIt;
        }
    }
    return { matchedIt, maxScore };
}
```
To cope with this specific issue and avoid any undefined behavior, I added an if statement to handle the case of d = 0 and modified the test case.

SkxPhan committed:
Fix Zero Division in case of perfect match
